project/authentication/models.py

```diff
@@ -2,7 +2,222 @@
 """
 Copyright (c) 2019 - present AppSeed.us
 """
+import enum
+from datetime import timezone
 
+from django.contrib.auth.base_user import AbstractBaseUser
+from django.contrib.auth.models import BaseUserManager
+from django.contrib.auth.models import PermissionsMixin
 from django.db import models
 
+
 # Create your domain models here.
+
+
+class RecordStatus(enum.Enum):
+    """
+    This enum class represents the status of every record in the database.
+    The default record status for new database entries is ACTIVE.
+    """
+    ACTIVE = "Active"
+    DELETED = "Delete"
+    ACTIVE_LOCKED = "Active Locked"
```

Comment:
@umarkayondo please add comments for each of these record states.

Reply:
@umarkayondo @deokyeyune That would be a good idea indeed. I am not sure how we will take that into account/enforce it: do we count on developers to not change entries which are marked as ACTIVE_LOCKED? Or should we try to implement it in code?